examples/trusted_database/module_0/rust/src/database.rs

```diff
@@ -0,0 +1,113 @@
+    last_update: String,
+    version: String,
+    #[serde(rename = "station", default)]
+    stations: Vec<Station>,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+struct Station {
+    id: u32,
+    name: String,
+    #[serde(rename = "terminalName", default)]
+    terminal_name: String,
+    #[serde(rename = "lat", default)]
+    latitude_degrees: f32,
+    #[serde(rename = "long", default)]
+    longitude_degrees: f32,
+    installed: bool,
+    locked: bool,
+    #[serde(rename = "installDate", default)]
+    install_date: String,
+    #[serde(rename = "removalDate", default)]
+    removal_date: String,
+    temporary: bool,
+    #[serde(rename = "nbBikes", default)]
+    number_of_bikes: u32,
+    #[serde(rename = "nbEmptyDocks", default)]
+    number_of_empty_docks: u32,
+    #[serde(rename = "nbDocks", default)]
+    number_of_docks: u32,
+}
+
+/// Load an XML database from [`oak::ReadHandle`] and parse it.
+pub fn load_database(config_map: ConfigMap) -> Result<Vec<PointOfInterest>, OakError> {
+    debug!("Loading database");
+    match config_map.items.get("database") {
+        Some(xml_database) => {
+            let points_of_interest = parse_database(xml_database).map_err(|error| {
+                error!("Couldn't parse database: {:?}", error);
+                OakError::OakStatus(oak_abi::OakStatus::ErrInvalidArgs)
+            })?;
+            debug!("Database loaded - size: {:?}", points_of_interest.len());
+            Ok(points_of_interest)
+        }
+        None => {
+            error!("`database` configuration argument is not specified");
+            Err(OakError::OakStatus(oak_abi::OakStatus::ErrInvalidArgs))
+        }
+    }
+}
+
+/// Parse an XML database into a vector of [`PointOfInterest`].
+pub fn parse_database(xml_database: &[u8]) -> Result<Vec<PointOfInterest>, OakError> {
+    let database: Database = from_str(
+        String::from_utf8(xml_database.to_vec())
+            .map_err(|error| {
+                error!("Couldn't convert vector to string: {:?}", error);
+                OakError::OakStatus(oak_abi::OakStatus::ErrInvalidArgs)
+            })?
+            .as_ref(),
+    )
+    .map_err(|error| {
+        error!("Couldn't parse XML data: {:?}", error);
+        OakError::OakStatus(oak_abi::OakStatus::ErrInvalidArgs)
+    })?;
+
+    let points_of_interest = database
+        .stations
+        .iter()
+        // Filter out uninstalled, closed and removed stations, and stations with no bikes.
+        .filter(|station| {
+            station.installed
+                && !station.locked
+                && station.removal_date.is_empty()
+                && station.number_of_bikes > 0
+        })
+        .map(|station| PointOfInterest {
+            name: station.name.to_string(),
+            location: Some(Location {
+                latitude: station.latitude_degrees,
+                longitude: station.longitude_degrees,
+            }),
+        })
```

Comment:
Could this be defined as `From` or `To` between these types?

Reply:
It's not necessarily `From`/`To` since it uses only 3 fields from `Station`